api/environments/models.py

```diff
@@ -430,13 +430,31 @@ def get_change_requests_metrics_queryset(self) -> QuerySet["ChangeRequest"]:
         return result
 
     def get_scheduled_metrics_queryset(self) -> QuerySet[FeatureState]:
-        result: QuerySet[FeatureState] = FeatureState.objects.filter(
-            environment=self,
-            identity_id__isnull=True,
-            feature_segment__isnull=True,
-            live_from__gt=timezone.now(),
+        from features.workflows.core.models import ChangeRequest
+
+        change_requests: QuerySet["ChangeRequest"] = (
+            ChangeRequest.objects.filter(
+                environment=self,
+                committed_at__isnull=False,
+                deleted_at__isnull=True,
+            )
+            .filter(
+                Q(
+                    Q(feature_states__deleted_at__isnull=True)
+                    & Q(feature_states__live_from__gt=timezone.now())
+                )
+                | Q(
+                    Q(environment_feature_versions__deleted_at__isnull=True)
+                    & Q(environment_feature_versions__live_from__gt=timezone.now())
+                )
+                | Q(
+                    Q(change_sets__deleted_at__isnull=True)
+                    & Q(change_sets__live_from__gt=timezone.now())
+                )
+            )
+            .distinct()
```

Comment:
Ah, distinct is very painful to deal with. Can we do without it? Maybe filter the queryset ourselves as a last resort?